Remotion/Web/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriter.cs

```diff
@@ -0,0 +1,126 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Web.UI;
+using Remotion.Utilities;
+using Remotion.Web.Contracts.DiagnosticMetadata;
+using Remotion.Web.UI;
+using Remotion.Web.UI.Controls.Rendering;
+
+namespace Remotion.Web.ContentSecurityPolicy
+{
+  /// <summary>
+  /// Represents <see cref="HtmlTextWriter"/> which implements Content-Security-Policy
+  /// </summary>
+  public class CspEnabledHtmlTextWriter : HtmlTextWriter
+  {
+    private static readonly IReadOnlyDictionary<string, string> s_supportedEvents =
+        new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+          { "onclick", "onclick" },
+          { "onchange", "onchange" },
+          { "oncontextmenu", "oncontextmenu" },
```

Comment:
```suggestion
          // can be removed once WebTreeView learns to use modern html script registration.
          { "oncontextmenu", "oncontextmenu" },
```

Since (or until) we don't add all the events at this time, we should note that we only have the contextmenu because changing remotion right now is too expensive. all the others are there because of WebForms (and then because of re-motion).